### ISSUE TITLE
Wildcard imports

### ISSUE THREAD

**Issue by nidunc** [mhoek2/tcp-share#4]
At the moment, wildcard imports are used throughout the project (for `tkinter`, among other things); here are some examples: https://github.com/mhoek2/tcp-share/blob/ae2fb85de0b6afa55f3ef576b710cc5ccfcb665e/main.py#L2 https://github.com/mhoek2/tcp-share/blob/ae2fb85de0b6afa55f3ef576b710cc5ccfcb665e/test_application.py#L8-L12  
This is convenient and saves one from having to type extra characters, but I’ve learned (through my code linter/formatter throwing warnings) that it is generally bad practice to do so[^1].

I suggest to at least think about our usage of those wildcard import, and whether it would be better (albeit less convenient) to use one of the following methods (code examples derived from [`/modules/gui/create_files` in my fork][github]):
- `import …`[^2]:
    ```python
    import tkinter
    
    header = tkinter.Label(self, text=f"{prefix}{i}")
    textbox = tkinter.Text(self, height=6, width=40)
    ```
- `import … as …`[^3]:
    ```python
    import tkinter as tk
    
    header = tk.Label(self, text=f"{prefix}{i}")
    textbox = tk.Text(self, height=6, width=40)
    ```
- `from … import …`[^4]:
    ```python
    from tkinter import Label, Text
    
    header = Label(self, text=f"{prefix}{i}")
    textbox = Text(self, height=6, width=40)
    ```

---

Relevant links:
- [Official Python documentation][pythonDocs1]
- [This question on Stack Overflow][stackOverflow1]
  - [And this question too][stackOverflow2]
- [This post on Reddit (/r/learnpython)][reddit1]
  - [And this one too (/r/learnprogramming)][reddit2]

[^1]: Partly because it imports everything from the module into the same namespace as the project, possibly overriding existing functions/constants/etc. (also overwriting prior imports).  
There are some exceptions to this ‘rule’, though (some examples in the reference links above), so it isn’t like we absolutely _have_ to change the wildcard imports. (And as far as overwriting existing imports, [that seems to be one ‘valid’ approach for `ttk`][pythonDocs2]—a more modern-looking implementation of some of `tkinter`s widgets.)
[^2]: The most hassle; one would have to write the entire module’s name every single time.
[^3]: Can be a lot more efficient, depending on how short the alias is compared to the module’s name.
[^4]: When actually using stuff from the imported modules, it looks identical to what one would use with wildcard imports. It can get a bit unruly when using many different things from the same module, though.

  [github]: https://github.com/nidunc/tcp-share/blob/b4a44668910217eee4ff6f7b7045cd47d66c67a7/modules/gui/create_files.py "tcp-share/modules/gui/create_files.py at master · nidunc/tcp-share"
  [pythonDocs1]: https://docs.python.org/3/tutorial/modules.html#importing-from-a-package/#:~:text=Although%20certain%20modules%20are%20designed%20to%20export%20only%20names%20that%20follow%20certain%20patterns%20when%20you%20use%20import%20*,%20it%20is%20still%20considered%20bad%20practice%20in%20production%20code "section 6.4.1. Importing * From a Package on 6. Modules — Python 3.13.0 documentation"
  [stackOverflow1]: https://stackoverflow.com/questions/3615125/should-wildcard-import-be-avoided "python - Should wildcard import be avoided? - Stack Overflow"
  [stackOverflow2]: https://stackoverflow.com/questions/2386714/why-is-import-bad "python - Why is \"import *\" bad? - Stack Overflow"
  [reddit1]: https://www.reddit.com/r/learnpython/comments/8zej2c/eli5_what_are_wildcard_imports_and_why_should_i/ "ELI5: What are wildcard imports and why should I avoid them? : r/learnpython"
  [reddit2]: https://www.reddit.com/r/learnprogramming/comments/3asmmd/difference_between_from_tkinter_import_import/ "difference between from Tkinter import */import Tkinter as tk? : r/learnprogramming"
  [pythonDocs2]: https://docs.python.org/3/library/tkinter.ttk.html#using-ttk/#:~:text=To%20override%20the%20basic%20Tk%20widgets,%20the%20import%20should%20follow%20the%20Tk%20import: "section Using Ttk on tkinter.ttk — Tk themed widgets — Python 3.13.0 documentation"

**Comment by mhoek2** [comment]
Very well documented issue.
I fully agree that this should be optimized.

However, content-related;
I believe that addressing this is more appropiate for a matured codebase.

During initial iterative development we have to prioritize core functionality and requirements over optimizing minor syntax styling.
Ofc, we should remain mindful of clear syntax, semantics, and conventions to reduce future refactoring.

syntax is still temporal, maintaining strict adherence to the described aspects too early is inefficient.
This mindset typically shifts during the first refactoring phase as the codebase evolves. We should avoid calling it 'convenient', that would apply if it was permanent.

The first refactoring phase is the right opportunity to clean up the codebase.
This is the appropriate time to address and apply consistency in semantics, syntax, docstrings and conventions.
Import conventions can be applied with greater certainty at that point, because the used module parts are determined.

To summarize:
It's a valuable learning experience in practicing GitHub issue filing. 
However, content-related; Given the codebase's small footprint, it’s more efficient to start thinking about these optimizations during the refactoring phase.
In this particular case, writing a detailed issue can take as much time, if not more, than revising the actual code itself.
